client.go

```diff
@@ -19,11 +21,14 @@ func NewClient(xp Transporter, u ErrorUnwrapper) *Client {
 // the result field will be populated (if applicable). It returns an Error
 // on error, where the error might have been unwrapped from Msgpack via the
 // UnwrapErrorFunc in this client.
-func (c *Client) Call(method string, arg interface{}, res interface{}) (err error) {
+func (c *Client) Call(ctx context.Context, method string, arg interface{}, res interface{}) (err error) {
 	var d dispatcher
 	go c.xp.Run()
 	if d, err = c.xp.getDispatcher(); err == nil {
-		err = d.Call(method, arg, res, c.errorUnwrapper)
+		if ctx == nil {
```

Comment:
I don't think we should check for nil -- we're going to have to touch all the call sites anyway, and so we can just use `context.TODO()` for now and fix up later